designs/2025-bulk-suppressions-api/README.md

```diff
@@ -0,0 +1,301 @@
+- Repo: eslint/eslint
+- Start Date: 2025-05-06
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: [Kentaro Suzuki](https://github.com/sushichan044)
+
+# Add support for bulk suppressions in `ESLint` and `LegacyESLint` classes
+
+## Summary
+
+<!-- One-paragraph explanation of the feature. -->
+
+This RFC proposes extending the bulk suppressions feature, introduced in ESLint 9.24.0 for CLI usage, to be available through the Node.js API via the `ESLint` and `LegacyESLint` classes.
+
+## Motivation
+
+<!-- Why are we doing this? What use cases does it support? What is the expected
+outcome? -->
+
+Currently, the bulk suppression feature introduced in ESLint 9.24.0 is only available via the CLI.
+This leads to inconsistencies when ESLint is used programmatically via its Node.js API, such as in IDE integrations.
+
+This leads to inconsistencies when ESLint is used programmatically. Violations suppressed using `eslint-suppressions.json` (especially when using a custom location via the CLI) might not be recognized when using the Node.js API, leading to incorrect error reporting in environments like IDEs.
```

Comment:
Suppose someone has a task to fix previously suppressed violations. How will they do this, since the violations no longer appear in their IDE?

Reply:
The motivation for using bulk suppression may be “I want to ignore existing errors and give priority to activating new errors”.
With that assumption, existing errors that should be ignored should not be shown in the IDE either.

To work on reducing the number of ignored errors, we can start by check the suppressed errors in `eslint-suppressions.json`.

Reply:
@mdjermanovic I think this is the same use case as when running on the CLI. Because the CLI automatically picks up the `eslint-suppressions.json` file, it's necessary to edit, move, or delete the file if your task is to clean up suppressed violations.

Reply:
I definitely _want_ to see the existing suppressions in my IDE, but I'd like to see them in a different color.

Reply:
@wagenet 
If that's the case, then we're probably thinking the same thing. Sorry for the unclear explanation.